trpc-case-study.md

```diff
@@ -0,0 +1,275 @@
+
+![alt-image](tRPC-media/database-result.png)
+
+## Strengths
+
+#### tRPC is lightweight
+Since tRPC is a protocol or method for accessing backend functionality, it allows for better performing applications without any code generation or runtime bloat.
+
+Since tRPC uses Typescript, which is a statically typed language, type definitions are checked during compiling rather than runtime, increasing speed of application.
+
+#### tRPC uses Typescript
+This can be viewed as both a strenght and a weakness. On one hand, it allows for a uniform coding language and infrastructure between backend and frontend, improving worklflow and communcation between the team. 
+
+While on the other hand, if Typescript is not the preferred programming language, or if team structure is heavily silo'd between front- and backend tRPC might not be the best option.
+
+#### tRPC works great in a monorepo
+With how tRPC communicates between client and server code, you are encouraged to use a singluar repository, and several highlighted features are only accessible if you do so. On a smaller scale, there are many benefits in doing so. It's easier to navigate between client/server code, making changes and ensure typesafety within the entirety of your code base. No more ``uncaught type-errors``.
+
+#### Quick development
+- Fastest and easiest way to develop an API. 
+- Uses type inferference, which means the type of data is automatically detected, which particularly boosts development in the early stages.
+- Integrate with IDE's.
+
+With some familiary to tRPC and well synergized techonologies, such as Next.js and Prisma, you can have functional endpoints up and running within minutes using a pre-configured project package. Such as [T3-app](https://create.t3.gg/).
+
+## Weaknesses
+
+#### tRPC uses Typescript
+While your front-end certainly has knowledge of Javascript, and possibly also well versed in Typescript. The same might not be the case for your backend team, or it might simply not be the preferred techonology. There is a greater variety in programming lanugages typically used for backend than there is for frontend. 
+
+While strict static typing leaves a lot of benefits, it can be frustrating to and cumbersome to learn, especially for junior level developers.
+
+#### Works best in a monorepo
+Since many of tRPC's features benefit from using a singular repository, it might not be the best option if the development team has a need or requirement for heavily silo'd frontend and backend.
+
+#### Scalability
+- In larger scale projects, in a larger team, its often beneficial to come up with a clear set of rules and a schema that everyone can work from. tRPC does not offer this.
+- If your app is getting increasingly more complex and require more nuanced requirements, tRPC might not be the best option.
```

Comment:
Why not?

Reply:
I've resolved this by explaining that other options provide better solutions for increasingly scaling complextion. So while there is nothing wrong with using tRPC in this scenario. The project might benefit more from a different solution.